### ISSUE TITLE
All .cs files have problems - every .cs file is red (red swigglies everywhere)

### ISSUE THREAD

**Issue by chickenSandwich** [OmniSharp/omnisharp-vscode#3976]
## Issue Description ##
All .cs files are reporting problems; for instance: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)

This occurred after an update of dotnet-runtime-3.1:amd64 (3.1.6-1, 3.1.7-1)
Other possibly relevant updates:
dotnet-runtime-2.1:amd64 (2.1.20-1, 2.1.21-1),
dotnet-runtime-3.1:amd64 (3.1.6-1, 3.1.7-1), 
dotnet-apphost-pack-3.1:amd64 (3.1.6-1, 3.1.7-1), 
dotnet-sdk-2.1:amd64 (2.1.808-1, 2.1.809-1), 
dotnet-sdk-3.1:amd64 (3.1.302-1, 3.1.401-1), 
dotnet-host:amd64 (3.1.6-1, 3.1.7-1), 
dotnet-hostfxr-2.1:amd64 (2.1.20-1, 2.1.21-1),
dotnet-hostfxr-3.1:amd64 (3.1.6-1, 3.1.7-1),
dotnet-runtime-deps-2.1:amd64 (2.1.20-1, 2.1.21-1),
dotnet-runtime-deps-3.1:amd64 (3.1.6-1, 3.1.7-1),
aspnetcore-runtime-2.1:amd64 (2.1.20-1, 2.1.21-1),
aspnetcore-runtime-3.1:amd64 (3.1.6-1, 3.1.7-1),
nodejs:amd64 (14.7.0-1nodesource1, 14.8.0-1nodesource1), 

## Steps to Reproduce ##
Create/Open any csharp project folder with VS Code

## Expected Behavior ##
Only report valid problems

## Actual Behavior ##
Problems are reported where there are none.  The project builds successfully.
- Restarting VSCode doesn't help
- Uninstalling/Re-installing C# extension doesn't help

### OmniSharp log ###
<details>Starting OmniSharp server at 8/13/2020, 7:40:41 AM
    Target: /Workspace/MyProject

OmniSharp server started with Mono 6.10.0.
    Path: /home/user/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/OmniSharp.exe
    PID: 31628

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on ubuntu 18.4 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 16.4 - "/usr/lib/mono/msbuild/Current/bin"
            2: StandAlone 16.4 - "/home/user/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 16.4 - "/usr/lib/mono/msbuild/Current/bin"
            CscToolExe = csc.exe
            CscToolPath = /home/user/.vscode/extensions/ms-dotnettools.csharp-1.22.1/.omnisharp/1.35.3/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Workspace/MyProject'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/Workspace/MyProject'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Workspace/MyProject/MyProject.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Workspace/MyProject'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Workspace/MyProject/MyProject.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Workspace/MyProject' on host 31491.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Workspace/MyProject/MyProject.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Workspace/MyProject/MyProject.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: MyProject
</details>

### C# log ###
<details>
The C# log is empty
</details>

## Environment information ##

**VSCode version**: 1.47.3
**C# Extension**: 1.22.1

<details><summary>Mono Information</summary>
        OmniSharp using global mono :6.10.0</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    39d17847db

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.401/

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  2.1.809 [/usr/share/dotnet/sdk]
  2.2.402 [/usr/share/dotnet/sdk]
  3.0.103 [/usr/share/dotnet/sdk]
  3.1.401 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.21 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|beautify|HookyQR|1.5.0|
|cpptools|ms-vscode|0.29.0|
|csharp|ms-dotnettools|1.22.1|
|csharpextensions|jchannon|1.3.1|
|docomment|k--kato|0.1.18|
|gitignore|codezombiech|0.6.0|
|glitch|glitch|0.1.6|
|indent-rainbow|oderwat|7.4.0|
|json-tools|eriklynd|1.0.2|
|rainbow-brackets|2gua|0.0.6|
|rainbow-csv|mechatroner|1.7.1|
|shader|slevesque|1.1.4|
|vscode-glsllint|CADENAS|1.1.1|
|vscode-nuget-package-manager|jmrog|1.1.6|
|whitespace|sandcastle|0.0.5|
|xml|DotJoshJohnson|2.5.1|;

</details>

**Comment by chickenSandwich** [comment]
This issue appears to be related to the version of node that is installed on my machine: (14.8.0).
Here is how I fixed this issue:

  - Checked version of node: 
`node -v`
`v14.8.0`
  - Installed n using npm:
`sudo npm install -g n`
  - Set node to use the latest stable version:
`sudo n stable`
  - Check version of node:
`node -v`
'v12.18.3'

**Comment by chickenSandwich** [comment]
I'm sorry, I spoke too soon.  This did not fix this issue.

**Comment by chickenSandwich** [comment]
I disabled all extensions except for C# 1.22.1
The log (Extension Host) has an error that might be helpful:

[2020-08-13 07:40:43.584] [exthost] [error] [ms-dotnettools.csharp] provider FAILED
[2020-08-13 07:40:43.584] [exthost] [error] Problem invoking 'GetCodeActions' on OmniSharp server: Error: Pending request cancelled: /v2/getcodeactions

**Comment by michellamin** [comment]
Had the same issue. Aparently the sdk was working just fine, since a could build my projects via CLI while vscode couldn't resolve my dependencies.

Had to rollback the sdk version e set it in the project's gloabl.json file.

1 - Download past release https://dotnet.microsoft.com/download/dotnet-core/3.1
2 - Move the contents from /dotnet/sdk/3.1.107/ to /usr/share/dotnet/sdk/ (Ubuntu)
3 - Set previous version o global.json file:
`{
  "sdk": {
    "version": "3.1.107"
  }
}`

**Comment by chickenSandwich** [comment]
Thank-you.  That was "painful", but it worked.  I ended up:
  - Downloading dotnet sdk 3.1.302
  - And moving the contents (as you suggested)
  - I didn't have a global.json in my project(s), so I added it.
    - At first I made a mistake and used : `{ "sdk": { "version": "3.1.401" } }` - which is the latest version installed on my machine.
    - It didn't work, and I thought: bummer :( 
    - But then I realized and corrected the version to: 3.1.302 and it worked.  I waited ... (because my machine is pretty slow), and I am so happy to report that worked.  Thank-you so much.  I hope that at some point they fix/update omnisharp for this issue.

**Comment by JoeRobich** [comment]
Issues with using the latest .NET Core SDK with our included Mono should be resolved in this pre-release - https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.23.0

**Comment by solrevdev** [comment]
Even with [https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.23.0/csharp-1.23.0.vsix](https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.23.0/csharp-1.23.0.vsix) I too am getting issues since .NET Core SDK was updated to `3.1.401`

mono version `Mono JIT compiler version 6.10.0.104 (2019-12/5d03a6fe116 Wed Apr 29 20:51:09 EDT 2020)`

![screenshot 2020-08-14 at 11 35 15](https://user-images.githubusercontent.com/374863/90241076-41a7ed00-de22-11ea-81ff-ccd20b00a58d.png)

omnisharp log 

```text
Starting OmniSharp server at 8/14/2020, 11:26:53 AM
    Target: /Users/solrevdev/Projects/myproject/myproject.sln

OmniSharp server started with Mono 6.10.0.
    Path: /Users/solrevdev/.vscode-insiders/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/OmniSharp.exe
    PID: 32899

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.14.6 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 16.8.0 - "/Library/Frameworks/Mono.framework/Versions/6.10.0/lib/mono/msbuild/Current/bin"
            2: StandAlone 16.8.0 - "/Users/solrevdev/.vscode-insiders/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 16.8.0 - "/Library/Frameworks/Mono.framework/Versions/6.10.0/lib/mono/msbuild/Current/bin"
            CscToolExe = csc.exe
            CscToolPath = /Users/solrevdev/.vscode-insiders/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/solrevdev/Projects/myproject'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        Detecting projects in '/Users/solrevdev/Projects/myproject/myproject.sln'.
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/solrevdev/Projects/myproject/src/emailparser/emailparser.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/solrevdev/Projects/myproject/src/instagrabber/instagrabber.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/solrevdev/Projects/myproject/src/web/web.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/solrevdev/Projects/myproject'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/solrevdev/Projects/myproject/src/emailparser/emailparser.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/solrevdev/Projects/myproject' on host 32254.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Users/solrevdev/Projects/myproject/src/emailparser/emailparser.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Users/solrevdev/Projects/myproject/src/emailparser/emailparser.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/solrevdev/Projects/myproject/src/instagrabber/instagrabber.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Users/solrevdev/Projects/myproject/src/instagrabber/instagrabber.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Users/solrevdev/Projects/myproject/src/instagrabber/instagrabber.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/solrevdev/Projects/myproject/src/web/web.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Users/solrevdev/Projects/myproject/src/web/web.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Users/solrevdev/Projects/myproject/src/web/web.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: emailparser
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: instagrabber
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: web
[info]: OmniSharp.Roslyn.CSharp.Services.Diagnostics.CSharpDiagnosticWorkerWithAnalyzers
        Solution initialized -> queue all documents for code analysis. Initial document count: 235.

```

dotnet info 

```text

 dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    39d17847db

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.401/

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  3.1.401 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

system 

```text
olrevdev@macmini
 OS: 64bit Mac OS X 10.14.6 18G95
 Kernel: x86_64 Darwin 18.7.0
 Uptime: 19d 22h 39m
 Packages: 468
 Shell: zsh 5.8
 Resolution: 2560x1440
 DE: Aqua
 WM: Quartz Compositor
 WM Theme: Blue (Dark)
 Disk: 374G / 525G (73%)
 CPU: Intel Core i5-4278U @ 2.60GHz
 GPU: Intel Iris
 RAM: 8590MiB / 16384MiB
```

**Comment by filipw** [comment]
this is caused by the old Mono on your machine that is no longer compatible with the latest dotnet SDK.
you can set `"omnisharp.useGlobalMono"` to `"never"` in VS code settings.

**Comment by solrevdev** [comment]
> this is caused by the old Mono on your machine that is no longer compatible with the latest dotnet SDK.
> you can set `"omnisharp.useGlobalMono"` to `"never"` in VS code settings.

@filipw Thank you so much that has fixed it for me 🙏

**Comment by alexthornton1** [comment]
I'm in the same boat on this one (on Ubuntu 20.04), regardless of what solution I open.  I've done the following so far:

- Installed the pre-release 1.23.0 C# extension linked by @JoeRobich above.
- Added the `"omnisharp.useGlobalMono": "never"` setting (and appear to have the same version of Mono, 6.10.0.104, installed locally that's in the OmniSharp installation) suggested by @filipw.

Unfortunately, the same issue persists.  Any other ideas on what to try next, aside from downgrading to the previous version of .NET Core SDK?

`dotnet --info`

```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    39d17847db

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.401/

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  3.1.401 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

OmniSharp log:

```
Starting OmniSharp server at 8/14/2020, 3:04:07 PM
    Target: /home/thornton/rdev/dotnet/Playground/Playground.sln

OmniSharp server started with Mono 6.10.0.
    Path: /home/thornton/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/OmniSharp.exe
    PID: 10462

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on ubuntu 20.4 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 16.8.0 - "/usr/lib/mono/msbuild/Current/bin"
            2: StandAlone 16.8.0 - "/home/thornton/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 16.8.0 - "/usr/lib/mono/msbuild/Current/bin"
            CscToolExe = csc.exe
            CscToolPath = /home/thornton/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/home/thornton/rdev/dotnet/Playground'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        Detecting projects in '/home/thornton/rdev/dotnet/Playground/Playground.sln'.
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/home/thornton/rdev/dotnet/Playground/Playground/Playground.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/home/thornton/rdev/dotnet/Playground'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /home/thornton/rdev/dotnet/Playground/Playground/Playground.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/home/thornton/rdev/dotnet/Playground' on host 10368.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/home/thornton/rdev/dotnet/Playground/Playground/Playground.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/home/thornton/rdev/dotnet/Playground/Playground/Playground.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: Playground
```

**Comment by filipw** [comment]
your log suggests that global mono is still in use. double check the omnisharp.useGlobalMono setting and/or restart VS Code

**Comment by alexthornton1** [comment]
Ah!  I had the setting in a `settings.json` file local to the solution, but it needed to be the global setting.  Back in business.  Thanks!

**Comment by filipw** [comment]
This has been resolved in 1.23.1
